cabal-install/tests/UnitTests/Distribution/Solver/Modular/DSL.hs

```diff
@@ -478,7 +478,7 @@ exAvSrcPkg ex =
           -- solver allows unknown extensions/languages when the compiler
           -- supports them.
           let checks = C.checkPackage (srcpkgDescription package) Nothing
-          in filter (not . isUnknownLangExt) checks
+          in filter (not . isMissingUpperBound) $ filter (not . isUnknownLangExt) checks
```

Comment:
Could this be expressed with an `&&` inside the filter?
Why do we need to filter `MissingUpperBound` here?

Reply:
A whole host of tests started failing because the generated package crashed on the check call in the generator, for example:
```
      show conflicts from final conflict set after exhaustive search:                                                                  FAIL
        Exception: invalid GenericPackageDescription for package A-1.0.0: [These packages miss upper bounds 'B','F' please add them using `cabal gen-bounds` for suggestions. For more information see:  https://pvp.haskell.org/]
        CallStack (from HasCallStack):
          error, called at tests/UnitTests/Distribution/Solver/Modular/DSL.hs:484:13 in main:UnitTests.Distribution.Solver.Modular.DSL
        Use -p '/show conflicts from final conflict set after exhaustive search/' to rerun this test only.
      show first conflicts after inexhaustive search:                                                                                  FAIL
        Exception: invalid GenericPackageDescription for package A-1.0.0: [These packages miss upper bounds 'B','F' please add them using `cabal gen-bounds` for suggestions. For more information see:  https://pvp.haskell.org/]
        CallStack (from HasCallStack):
```
filtering out the missing upper bound warning here makes the test work as intended.

Reply:
Makes sense.

Reply:
I see, but then why not just one filter instead of two? `\x -> (not . isSomething) x && (not . isSomethingElse x)`

Reply:
That's outdated code. This is just one filter now.